backend/src/v5/processors/teamspaces/projects/models/drawings.js

```diff
@@ -35,13 +37,41 @@ Drawings.getDrawingList = async (teamspace, project, user) => {
 	return getModelList(teamspace, project, user, modelSettings);
 };
 
+Drawings.getDrawingStats = async (teamspace, drawing) => {
+	let latestRev;
+	const [settings, revCount] = await Promise.all([
+		getDrawingById(teamspace, drawing, { number: 1, status: 1, type: 1, desc: 1, calibration: 1 }),
+		getRevisionCount(teamspace, drawing, modelTypes.DRAWING),
+	]);
+
+	try {
+		latestRev = await getLatestRevision(teamspace, drawing, modelTypes.DRAWING,
+			{ statusCode: 1, revCode: 1, timestamp: 1 });
+	} catch {
+		// do nothing. A drawing can have 0 revision.
+	}
+
+	return {
+		number: settings.number,
+		status: settings.status,
+		type: settings.type,
+		desc: settings.desc,
+		revisions: {
+			total: revCount,
+			lastUpdated: latestRev?.timestamp,
+			latestRevision: latestRev ? `${latestRev.statusCode}-${latestRev.revCode}` : undefined,
+		},
+		calibration: settings.calibration ?? 'uncalibrated',
+	};
+};
+
 Drawings.addDrawing = (teamspace, project, data) => addModel(teamspace, project,
 	{ ...data, modelType: modelTypes.DRAWING });
 
 Drawings.updateSettings = updateModelSettings;
 
 Drawings.deleteDrawing = async (teamspace, project, drawing) => {
-	await removeFilesWithMeta(teamspace, DRAWINGS_HISTORY_REF_COL, { model: drawing });
+	await removeFilesWithMeta(teamspace, `${modelTypes.DRAWING}s.history.ref`, { model: drawing });
```

Comment:
you don't need to append `.ref` to the collection name

Reply:
should probably declare a constant in the models constant file like you had before...